### PR TITLE
feat: pl-file-editor allow blank

### PR DESCRIPTION
**Pull request by eliotwrobson**
See title, in response to #6350, adds the allow blank option to the pl-file-editor element. Had issues testing locally because of external grader issues, so someone else will have to test, but I think this will work correctly.

### CODE REVIEW

**Comment by jonatanschroeder** [comment]
I did a quick check and indeed it seems the file is not created, but there is an issue: once submitted as blank, the editor defaults back to the initial code. That's because the code treats the lack of a file as "there was never a submission", which is not the case here.

So there would need to be a distinction between "the submitted file does not exist because it was never submitted" and "the submitted file does not exist because the last submission did not include it". The former defaults to the initial content, the latter keeps the text blank. Not sure how to handle this case, though, @mwest1066 any thoughts?

**Comment by mwest1066** [comment]
> So there would need to be a distinction between "the submitted file does not exist because it was never submitted" and "the submitted file does not exist because the last submission did not include it". The former defaults to the initial content, the latter keeps the text blank. Not sure how to handle this case, though, @mwest1066 any thoughts?

Help me to understand the exact sequence here. Do you mean that a student edits the file, submits it, then clears the editor contents and submits again, thus resulting in no file in the second submission and the editor resetting to the initial content?

**Comment by jonatanschroeder** [comment]
> Help me to understand the exact sequence here. Do you mean that a student edits the file, submits it, then clears the editor contents and submits again, thus resulting in no file in the second submission and the editor resetting to the initial content?

There isn't a need for a second submission. In essence, get a `pl-file-editor` with some initial source code (e.g., the question used for testing in this PR), clear up the contents of the file, and submit it. The submission panel will correctly indicate that there is no submitted file, however the question panel will restore the file editor to the initial source code. This is inconsistent with the behaviour of these elements, which typically start the question panel with the value in the last submission for the current variant, which should be blank in this case.

**Comment by nwalters512** [comment]
@jonatanschroeder I believe I fixed this in `fbdb333`. In short, we previously weren't persisting anything if the submission was empty. Now we do, but only if `allow-empty="true"` is set.

File: apps/prairielearn/elements/pl-file-editor/pl-file-editor.py [hunk 1]
```diff
@@ -150,27 +151,22 @@ def render(element_html, data):
...
     answer_name = get_answer_name(file_name)
     normalize_to_ascii = pl.get_boolean_attrib(
         element, "normalize-to-ascii", NORMALIZE_TO_ASCII_DEFAULT
     )
+    allow_blank = pl.get_boolean_attrib(element, "allow-blank", ALLOW_BLANK_DEFAULT)
 
     # Get submitted answer or return parse_error if it does not exist
-    file_contents = data["submitted_answers"].get(answer_name, None)
-    if not file_contents:
+    file_contents = data["submitted_answers"].get(answer_name, "")
```
Comment (nwalters512) [inline apps/prairielearn/elements/pl-file-editor/pl-file-editor.py:168]: Changing the default to `""` was necessary for type-checking. I believe this is safe: if `allow-blank` is false, we'll just add an error and ignore the contents entirely. And if `allow-blank` is true, then we don't really care about the distinction between "not present" and "empty".

File: exampleCourse/questions/demo/autograder/python/numpy/question.html
Comment (jonatanschroeder) [inline exampleCourse/questions/demo/autograder/python/numpy/question.html:1]: While I'm ok with testing the feature in this question, I don't think this should go to the final version...

**Review by jonatanschroeder — COMMENTED** [review]
This is also missing a docs update.

**Comment by eliotwrobson** [comment]
@jonatanschroeder Updated the example question here: https://github.com/PrairieLearn/PrairieLearn/pull/8820/commits/9c5f2b8723c9bab22de679f094e4903dbff25abf

**Review by nwalters512 — APPROVED** [review]
LGTM, but since some of these were my changes, I'd like Jonatan to sign off as well.

**Review by jonatanschroeder — APPROVED** [review]

**Pull request merged**